arithmetization/src/test/java/net/consensys/linea/sequencer/txselection/MaxBlockBytesSizeTransactionSelectorTest.java

```diff
@@ -0,0 +1,115 @@
+import static org.hyperledger.besu.plugin.data.TransactionSelectionResult.SELECTED;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+
+import net.consensys.linea.sequencer.txselection.selectors.MaxBlockSizeTransactionSelector;
+import org.hyperledger.besu.datatypes.PendingTransaction;
+import org.hyperledger.besu.datatypes.Transaction;
+import org.hyperledger.besu.evm.log.Log;
+import org.hyperledger.besu.plugin.data.TransactionProcessingResult;
+import org.hyperledger.besu.plugin.data.TransactionSelectionResult;
+import org.hyperledger.besu.plugin.services.txselection.PluginTransactionSelector;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class MaxBlockBytesSizeTransactionSelectorTest {
+  private MaxBlockSizeTransactionSelector txSelector;
+
+  @BeforeEach
+  public void initialize() {
+    int transactionCount = 1;
+    int logsCount = 2;
+    int maxBytesPerBlock =
+        EMPTY_L1_BLOCK_SIZE
+            + (transactionCount * (FROM_ADDRESS_SIZE + AVERAGE_TX_RLP_SIZE))
+            + (logsCount * LOG_SIZE);
+    txSelector = new MaxBlockSizeTransactionSelector(maxBytesPerBlock);
+  }
+
+  @Test
+  public void shouldSelectWhen_BlockBytesSize_IsLessThan_MaxBytesPerBlock() {
+    var mockTransactionProcessingResult = mockTransactionProcessingResultWithLogs(1);
+    verifyTransactionSelection(
+        txSelector, mockTransaction(), mockTransactionProcessingResult, SELECTED);
+    assertThat(txSelector.getCumulativeBlockBytesSize())
+        .isLessThan(txSelector.getMaxBytesPerBlock());
+  }
+
+  @Test
+  public void shouldSelectWhen_BlockBytesSize_IsEqual_MaxBytesPerBlock() {
+    var mockTransactionProcessingResult = mockTransactionProcessingResultWithLogs(2);
+    verifyTransactionSelection(
+        txSelector, mockTransaction(), mockTransactionProcessingResult, SELECTED);
+    assertThat(txSelector.getCumulativeBlockBytesSize())
+        .isEqualTo(txSelector.getMaxBytesPerBlock());
+  }
+
+  @Test
+  public void shouldNotSelectWhen_BlockBytesSize_IsGreaterThan_MaxBytesPerBlock() {
+    var mockTransactionProcessingResult = mockTransactionProcessingResultWithLogs(3);
+    verifyTransactionSelection(
+        txSelector, mockTransaction(), mockTransactionProcessingResult, BLOCK_BYTES_SIZE_OVERFLOW);
+    assertThat(txSelector.getCumulativeBlockBytesSize())
+        .isLessThan(txSelector.getMaxBytesPerBlock());
+  }
+
+  private void verifyTransactionSelection(
+      final PluginTransactionSelector selector,
+      final PendingTransaction transaction,
+      final TransactionProcessingResult processingResult,
+      final TransactionSelectionResult expectedSelectionResult) {
+    var selectionResult = selector.evaluateTransactionPostProcessing(transaction, processingResult);
+    assertThat(selectionResult).isEqualTo(expectedSelectionResult);
+    notifySelector(txSelector, transaction, processingResult, selectionResult);
```

Comment:
The notify here does not really do anything because you are only calling the verifyTransactionSelection once in each of the tests.
You could add one, where you call the verifyTransactionSelection twice, where the second transaction exceeds the limit (but the second tx on it's own would have been OK).

Reply:
I need to notify the selector otherwise I cannot assert the cumulativeBlockSize value in the selector.